hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/CommonUtils.java

```diff
@@ -88,4 +138,82 @@ public static boolean productWouldOverflow(final long multiplier, final long mul
+        DataCase dataCase = txn.getDataCase();
+
+        return switch (dataCase) {
+            case CONTRACTCALL -> ContractCall;
+            case CONTRACTCREATEINSTANCE -> ContractCreate;
+            case CONTRACTUPDATEINSTANCE -> ContractUpdate;
+            case CONTRACTDELETEINSTANCE -> ContractDelete;
+            case ETHEREUMTRANSACTION -> EthereumTransaction;
+            case CRYPTOADDLIVEHASH -> CryptoAddLiveHash;
+            case CRYPTOAPPROVEALLOWANCE -> CryptoApproveAllowance;
+            case CRYPTODELETEALLOWANCE -> CryptoDeleteAllowance;
+            case CRYPTOCREATEACCOUNT -> CryptoCreate;
+            case CRYPTODELETE -> CryptoDelete;
+            case CRYPTODELETELIVEHASH -> CryptoDeleteLiveHash;
+            case CRYPTOTRANSFER -> CryptoTransfer;
+            case CRYPTOUPDATEACCOUNT -> CryptoUpdate;
+            case FILEAPPEND -> FileAppend;
+            case FILECREATE -> FileCreate;
+            case FILEDELETE -> FileDelete;
+            case FILEUPDATE -> FileUpdate;
+            case SYSTEMDELETE -> SystemDelete;
+            case SYSTEMUNDELETE -> SystemUndelete;
+            case FREEZE -> Freeze;
+            case CONSENSUSCREATETOPIC -> ConsensusCreateTopic;
+            case CONSENSUSUPDATETOPIC -> ConsensusUpdateTopic;
+            case CONSENSUSDELETETOPIC -> ConsensusDeleteTopic;
+            case CONSENSUSSUBMITMESSAGE -> ConsensusSubmitMessage;
+            case UNCHECKEDSUBMIT -> UncheckedSubmit;
+            case TOKENCREATION -> TokenCreate;
+            case TOKENFREEZE -> TokenFreezeAccount;
+            case TOKENUNFREEZE -> TokenUnfreezeAccount;
+            case TOKENGRANTKYC -> TokenGrantKycToAccount;
+            case TOKENREVOKEKYC -> TokenRevokeKycFromAccount;
+            case TOKENDELETION -> TokenDelete;
+            case TOKENUPDATE -> TokenUpdate;
+            case TOKENMINT -> TokenMint;
+            case TOKENBURN -> TokenBurn;
+            case TOKENWIPE -> TokenAccountWipe;
+            case TOKENASSOCIATE -> TokenAssociateToAccount;
+            case TOKENDISSOCIATE -> TokenDissociateFromAccount;
+            case TOKEN_FEE_SCHEDULE_UPDATE -> TokenFeeScheduleUpdate;
+            case TOKEN_PAUSE -> TokenPause;
+            case TOKEN_UNPAUSE -> TokenUnpause;
+            case SCHEDULECREATE -> ScheduleCreate;
+            case SCHEDULEDELETE -> ScheduleDelete;
+            case SCHEDULESIGN -> ScheduleSign;
+            case UTIL_PRNG -> UtilPrng;
+            default -> throw new UnknownHederaFunctionality("Unknown HederaFunctionality for " + txn);
+        };
+    }
+
+    /**
+     * check TransactionBody and return the NodeStakeUpdate HederaFunctionality
+     *
+     * @param txn the {@code TransactionBody}
+     * @return one of HederaFunctionality
+     * @throws UnknownHederaFunctionality if it's not a NodeStakeUpdate HederaFunctionality
+     */
+    @NonNull
+    public static HederaFunctionality functionOrStakeUpdateOf(@NonNull final TransactionBody txn)
```

Comment:
This change was from @tinker-michaelj 's comments and has to be treated differently. This is not a valid type of transaction for a user to submit via HAPI, and is added for the node software to use to export stake info to mirror nodes in the record stream. This could be applied to all other record stream. So it's here.